docs/design-documents/README.md

```diff
@@ -0,0 +1,40 @@
+# Design Process for Mbed-OS
+This document defines the software design process for Mbed-OS. Anyone developing for Mbed-OS should use this process to submit the design to get it reviewed from the broader Mbed-OS developer community.
```

Comment:
What does it mean "design" ? 

When to use this process? When I do large refactor or want to submit new feature or change user API? HAL change?

Reply:
Its up to the developer to make a judgement on whether a design doc is required for a change. For small changes its definitely not required. But for changes with lot of code turmoil its best to have a backing design document. Let me try to see if we can update the process document to capture this guideline.

Reply:
I think it's "Mbed OS" (as opposed to "Mbed-OS") ;)